Services/Exceptions/classes/class.ilSoapExceptionHandler.php

```diff
@@ -22,6 +22,10 @@ class ilSoapExceptionHandler extends \Whoops\Handler\Handler
 {
     private function buildFaultString(): string
     {
+        if (!defined('DEVMODE') || DEVMODE !== 1) {
+            return htmlspecialchars($this->getInspector()->getException()->getMessage());
```

Comment:
If the `DEVMODE` is *disaboed*, we only show the error message (on productive systems), *not* the complete stack trace (and optionally previous exceptions).

An error will be put to the log in any case (if `ini_get('log_errors')` is enabled for PHP), see: `\ilErrorHandling::loggingHandler`